assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -13366,6 +13426,34 @@ public void doModel_answer(RunData data) {
         }
     }
 
+    private void assignment_extension_option_into_context(Context context, SessionState state){
+        String allowExtensionTimeString = null;
+        if (state.getAttribute(AssignmentConstants.ALLOW_EXTENSION_CLOSETIME) != null){
+            allowExtensionTimeString = (String) state.getAttribute(AssignmentConstants.ALLOW_EXTENSION_CLOSETIME);
+            Instant extensionCloseTime = Instant.ofEpochMilli(Long.parseLong(allowExtensionTimeString));
+            context.put("extensionCloseTime", assignmentService.getUsersLocalDateTimeString(extensionCloseTime));
+        }
+        putTimePropertiesInContext(context, state, "Extension", ALLOW_EXTENSION_CLOSEMONTH, ALLOW_EXTENSION_CLOSEDAY, ALLOW_EXTENSION_CLOSEYEAR, ALLOW_EXTENSION_CLOSEHOUR, ALLOW_EXTENSION_CLOSEMIN);
+    }
+
+    private void assignment_extension_option_into_state(Assignment a, AssignmentSubmission s, SessionState state){
```

Comment:
Is it possible to add the AssignmentConstants.ALLOW_EXTENSION_CLOSETIME as a parameter vs adding to the session state?

Were try avoid adding anything new to the session state, as it's a bad pattern.

Reply:
Sorry, missed this message. I'm sure this doesn't need to be added to the session state. Myself or Joe will take a look at this as soon as possible.

Reply:
Hey Earle. I finally got around to taking a look at this today. It looks like the extension feature was written to match the existing pattern that resubmission uses. Would it make sense to leave this as it's written and open a new ticket to refactor both?

Reply:
yep